lib/netsuite/response.rb

```diff
@@ -16,5 +18,30 @@ def success!
     def success?
       @success
     end
+
+    private
+
+    def status_detail
+      @body &&
+        @body.is_a?(Hash) &&
+        @body[:status] &&
+        @body[:status][:status_detail]
+    end
+
+    def response_error_code
+      if success?
+        nil
+      else
+        status_detail &&
+          status_detail[:code]
+      end
+    end
+
+    def raise_on_response_errors
+      case response_error_code
+      when 'INSUFFICIENT_PERMISSION'
+        raise NetSuite::PermissionError, status_detail[:message]
+      end
```

Comment:
@aom what other error types do you think would be added to this switch statement?

Reply:
@iloveitaly I don't know yet :) By implementing switch-statement I wanted to embrace the possibility of handling more errors that should raise.

Reply:
https://netsuite.custhelp.com/app/answers/detail/a_id/11613 I see a lot of error types, but I don't know which will be useful here.

Reply:
It seems my user account doesn't have access to that link.

Reply:
Nevermind. I had to open SuiteAnswers from NetSuite Help so it authenticated me and I was able to open the link directly.

Reply:
Included list to the documentation as a comment in `raise_on_response_errors` method. 

In my use case I've burned myself with INSUFFICIENT_PERMISSION errors often enough that I wanted it to raise instead of fail silently. I can't say whether other error codes should raise but I assume someone might have a good case to add more of them to this list in future!

Reply:
@iloveitaly @aom This makes good sense to me. Though I would think that we wouldn't want to fail silently for any unsuccessful requests. 

Would it be useful to have a "generic" ResponseError raised for all other error types and then we could add to the list of specific errors as needed?

```
def raise_on_response_errors
  return unless response_error_code

  case response_error_code
  when 'INSUFFICIENT_PERMISSION'
    raise NetSuite::PermissionError, status_detail[:message]
  else 
    raise NetSuite::UnsuccessfulResponseError, "#{status_detail[:code]}: #{status_detail[:message]}"
  end
end

```

I don't know if it would be awkward to have have a mix of specific and generic errors raised

Reply:
Also, I realize that this PR has been open for a while. Adding the generic error could definitely be done in a second phase so it doesn't hold this one up any longer.